js/modules/k6/grpc/client.go

```diff
@@ -249,7 +243,7 @@ func (c *Client) Connect(ctxPtr *context.Context, addr string, params map[string
 	// returns. We only need to close the channel to un-block in a non-error scenario;
 	// otherwise it can be GCd without closing as we return on an error on the channel.
 	errc := make(chan error, 1)
-	go func() {
+	go func(state *lib.State) {
```

Comment:
nit: not particularly certain this adds anything -  both the `Client` and it's vu (`c.vu`) get accessed all the time so making it explicit for `state` seems strange. Is there something in particular this fixes?

Reply:
Yep, the linter

Reply:
do you mean 
```js/modules/k6/grpc/client.go:211:2: variable 'state' is only used in the if-statement (js/modules/k6/grpc/client.go:212:2); consider using short syntax (ifshort)
```

where actually ifshort is *wrong* ?

Reply:
which I think is https://github.com/esimonov/ifshort/issues/18

Reply:
yep, that is the case, so I would prefer to stay safe without using a `nolint` directive and refactor when the state will be used outside. I can add a comment to make it visible and clear. WDYT?

Reply:
mmm...I think we could refactor a bit and remove several problems with one shot, trying

Reply:
Done, applied a simple refactor so also the other nolint directives have been removed